standards/core/lightpush.md

```diff
@@ -0,0 +1,88 @@
+---
+title: WAKU2-LIGHTPUSH
+name: Waku v2 Light Push
+status: draft
+editor: Zoltan Nagy <zoltan@status.im> 
+contributors: 
+  - Hanno Cornelius <hanno@status.im>
+  - Daniel Kaiser <danielkaiser@status.im>
+  - Oskar Thorén <oskarth@titanproxy.com>
+---
+
+previous version: `/vac/waku/lightpush/2.0.0-beta1` [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/waku/standards/core/19/lightpush)
+
+---
+**Protocol identifier**: `/vac/waku/lightpush/3.0.0`
+
+## Motivation and Goals
+
+Light nodes with short connection windows and limited bandwidth wish to push messages to other nodes in the Waku network to request message services.
+A common use case is to request that service node to publish the message to a `11/WAKU2-RELAY` pubsub topic.
+Additionally, there is sometimes a need for confirmation that a message has been received "by the network"
+(here, at least one node).
+
+`WAKU2-LIGHTPUSH` is a request/response protocol for this.
+
+## Payloads
+
+```protobuf
+syntax = "proto3";
+
+message LightPushRequest {
+        string request_id = 1;
+    uin32 request_type = 10;  // reserved for future request type selection, currently it is always RELAY (0)
+    string topic = 20;
+    WakuMessage message = 21;
+}
+
+message LightPushResponse {
+    string request_id = 1;
+    uint32 status_code = 10; // non zero in case of failure, see appendix
+    optional string status_desc = 11;
+    uint32 relay_peer_count = 12;
+}
+```
+
+### Message Relaying
+
+Nodes that respond to `LightPushRequest` MUST either relay the encapsulated message via [11/WAKU2-RELAY](https://rfc.vac.dev/waku/standards/core/11/relay) protocol on the specified `pubsub_topic` or `shard` ([WAKU2-RELAY-SHARDING](https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md)) depending on the network configuration.
+If they are unable to do so for some reason, they SHOULD return an error code in `LightPushResponse`.
+
+### Examples of possible error codes
+
+| Result | Code | Note |
+|--------|------|------|
+| SUCCESS  | 0 | Successfull push, response's relay_peer_count holds the number of peers the message is pushed.    |
+| BAD_REQUEST | 400   |     |
+| NO_PEERS_TO_RELAY | 404 | Service node has no relay peers  |
```

Comment:
A possible suggestion because the `NO_PEERS_TO_RELAY` value has a connotation of zero peers.
```suggestion
| NOT_ENOUGH_PEERS_TO_RELAY | 404 | Service node has no relay peers  |
```

Reply:
I'd suggest sticking to HTTP codes as much as possible. A `404` usually implies an addressed resource was "not found" - a mistake the client needs to fix. Wouldn't "no peers to relay" match more closely to the "service unavailable" `503`? That is, you requested a service, but it is temporarily unavailable on the server-side due to lack of peers to relay to. We could in theory also use some of the unused HTTP codes to have more specific error cases.
Generally server-side errors should be in the `500` range.
Client-side errors (e.g. addressing a missing resource) should be in the `400` range.

See e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
Agree with @jm-clius and I think we should have: 
- SUCCESS - 200;
- BAD_REQUEST - to me is questionable, do we even need it? 
- UNSUPPORTED_TOPIC - maybe `421`?
- SERVICE_UNAVAILABLE overlaps with NO_PEERS_TO_RELAY - seems to me too;

Also, we need to figure out how to apply `418`! I always dreamt of doing so. 

cc @NagyZoltanPeter

Reply:
@weboko: I'm not sure that having non-zero success helps here, but I will do it to get it confirm with STORE's status/error codes.
418? :-D maybe for BAD_REQUEST, that's the closest, haha.

Reply:
Oh, and yes, BAD_REQUEST is anything if we cannot decode the request's payoad. It can be due to missused (wrong) protobuf definition.

Reply:
@weboko: For `SERVICE_UNAVAILABLE overlaps with NO_PEERS_TO_RELAY` question. I'm not pretty sure. There can be different reason why we don't have peers.
I'm thinking of: SERVICE_UNAVAILABLE - after startup we did not get connected to any peers yet, or we are disconnected for some reason. While NO_PEERS_TO_RELAY is rather related to the used pubsub_topic/shard we do not have connected peers that supports that. WDYT? I would keep both in this form.